ssl/quic/quic_wire_pkt.c

```diff
@@ -332,6 +336,8 @@ int ossl_quic_wire_decode_pkt_hdr(PACKET *pkt,
             if (!PACKET_forward(pkt, hdr->len))
                 return 0;
         } else if (version != QUIC_VERSION_1) {
+            if (fail_cause != NULL)
+                *fail_cause |= QUIC_PKT_HDR_DECODE_BAD_VERSION;
```

Comment:
I think this needs more thought here. This is the RFC 9000:
```
6.1.  Sending Version Negotiation Packets

   If the version selected by the client is not acceptable to the
   server, the server responds with a Version Negotiation packet; see
   Section 17.2.1.  This includes a list of versions that the server
   will accept.  An endpoint MUST NOT send a Version Negotiation packet
   in response to receiving a Version Negotiation packet.
```

Also section 6 says:
```
   The size of the first packet sent by a client will determine whether
   a server sends a Version Negotiation packet.  Clients that support
   multiple QUIC versions SHOULD ensure that the first UDP datagram they
   send is sized to the largest of the minimum datagram sizes from all
   versions they support, using PADDING frames (Section 19.1) as
   necessary.  This ensures that the server responds if there is a
   mutually supported version.  A server might not send a Version
   Negotiation packet if the datagram it receives is smaller than the
   minimum size specified in a different version; see Section 14.1.
```

putting these two together: QUIC server should silently discard packet if it is short and does not contain a valid quic header. if packet is large enough and contains a version number server does not support, then server needs to reply.

So we should perhaps have error DECODE error which means the packet should be silently discarded. anything else deserves a response from port.

Reply:
I think we (roughly) have this now.  that is to say, if ossl_quic_wire_decode_pkt_hdr() returns zero, and the fail_flags pointer contains no causal flags, then we discard the frame.  i.e. a failure in decoding without a causal flag is treated as a general DECODE error.  But we can certainly make that more positive by adding a general DECODE flag error cause.

Reply:
added a generic decode error failure clause